/**
 * Copyright 2019 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// ms_manager.proto
syntax = "proto3";

package mindspore.serving.proto;
import "mindspore_serving/proto/ms_service.proto";

message DistributedServableSpec {
  // servable name
  string name = 1;
  // optional. If unspecified, the latest version servable will be used.
  int64 version_number = 2;
  // Specifies the method name in the servable.
  string method_name = 3;
}

message DistributedPredictRequest {
  DistributedServableSpec servable_spec = 1;
}

message DistributedPredictReply {
  DistributedServableSpec servable_spec = 1;
  repeated ErrorMsg error_msg = 2;
}

message DistributedExitRequest {
  string address = 1;
}

message DistributedExitReply {
  ErrorMsg error_msg = 1;
}

service MSAgent {
  rpc DistributedPredict(DistributedPredictRequest) returns (DistributedPredictReply) {}
  rpc DistributedExit(DistributedExitRequest) returns (DistributedExitReply) {}
}
